examples/dvbsrc.c

```diff
@@ -0,0 +1,268 @@
+    struct uprobe *uprobe_main = uprobe_stdio_alloc(NULL, stderr, loglevel);
+    assert(uprobe_main != NULL);
+    uprobe_main = uprobe_uref_mgr_alloc(uprobe_main, uref_mgr);
+    assert(uprobe_main != NULL);
+    uprobe_main = uprobe_uclock_alloc(uprobe_main, uclock);
+    assert(uprobe_main != NULL);
+    uprobe_main = uprobe_ubuf_mem_pool_alloc(uprobe_main, umem_mgr,
+            UBUF_POOL_DEPTH, UBUF_SHARED_POOL_DEPTH);
+    assert(uprobe_main != NULL);
+    uprobe_main = uprobe_pthread_upump_mgr_alloc(uprobe_main);
+    assert(uprobe_main != NULL);
+    uref_mgr_release(uref_mgr);
+    uclock_release(uclock);
+    umem_mgr_release(umem_mgr);
+    uprobe_pthread_upump_mgr_set(uprobe_main, upump_mgr);
+
+    struct uprobe *uprobe_dejitter = uprobe_dejitter_alloc(uprobe_use(uprobe_main), true, 0);
+    assert(uprobe_dejitter != NULL);
+
+    struct upipe_mgr *src_xfer_mgr = upipe_pthread_xfer_mgr_alloc(255,
+            20, uprobe_use(uprobe_main), upump_ev_mgr_alloc_loop,
+            UPUMP_POOL, UPUMP_BLOCKER_POOL, NULL, NULL, NULL);
+    assert(src_xfer_mgr != NULL);
+    struct upipe_mgr *upipe_wsrc_mgr = upipe_wsrc_mgr_alloc(src_xfer_mgr);
+    assert(upipe_wsrc_mgr != NULL);
+    upipe_mgr_release(src_xfer_mgr);
+
+    uprobe_throw(uprobe_main, NULL, UPROBE_FREEZE_UPUMP_MGR);
+
+    /* dvb */
+    struct upipe_mgr *upipe_dvbsrc_mgr = upipe_dvbsrc_mgr_alloc();
+    struct upipe *upipe_src = upipe_void_alloc(upipe_dvbsrc_mgr,
+            uprobe_pfx_alloc(uprobe_use(uprobe_dejitter), loglevel , "dvbsrc"));
+    upipe_mgr_release(upipe_dvbsrc_mgr);
+    assert(upipe_src);
+    dvbsrc = upipe_src;
+
+    ubase_assert(upipe_set_uri(upipe_src, src));
+    ubase_assert(upipe_set_option(upipe_src, "sys", sys));
+    ubase_assert(upipe_set_option(upipe_src, "lnb", lnb));
+    ubase_assert(upipe_set_option(upipe_src, "frequency", freq));
+    ubase_assert(upipe_set_option(upipe_src, "symbol-rate", srate));
+    ubase_assert(upipe_set_option(upipe_src, "inner-fec", fec));
+    ubase_assert(upipe_set_option(upipe_src, "polarization", pol));
+    ubase_assert(upipe_attach_uclock(upipe_src));
+
+    uprobe_throw(uprobe_main, NULL, UPROBE_THAW_UPUMP_MGR);
+
+    /* deport to the source thread */
+    upipe_src = upipe_wsrc_alloc(upipe_wsrc_mgr,
+            uprobe_pfx_alloc(uprobe_use(uprobe_main),
+                UPROBE_LOG_VERBOSE, "wsrc"),
+            upipe_src,
+            uprobe_pfx_alloc(uprobe_use(uprobe_main),
+                UPROBE_LOG_VERBOSE, "wsrc_x"),
+            1000);
+
+    upipe_mgr_release(upipe_wsrc_mgr);
+
+    struct upipe_mgr *udpsink_mgr = upipe_udpsink_mgr_alloc();
+    struct upipe *udpsink = upipe_void_alloc_output(upipe_src, udpsink_mgr,
+            uprobe_pfx_alloc(uprobe_use(uprobe_main), loglevel, "udp sink"));
+    upipe_mgr_release(udpsink_mgr);
+    assert(udpsink);
+    ubase_assert(upipe_attach_uclock(udpsink));
+    upipe_release(udpsink);
+
+    ubase_assert(upipe_set_uri(udpsink, dst));
+
+    pthread_t p;
```

Comment:
I guess at least a comment is needed for the approach chosen.

Reply:
While I understand the concern for latency, I deeply dislike calling a control function of a pipe from a different thread. Does libdvb even support being called from two different threads? What if upipe_set_uri is called while you're retrieving the stats? I think this is wrong.

Reply:
It's not latency, it's the control taking so long that udp packets are late.

Reply:
Another way would be to have a control function that returns the frontend file descriptor, so you can directly call the ioctl. Or have a manager control command do it.